src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+     * @throws IOException launched if a problem exists reading the PEM information
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
+     */
+    public PEMManager(@Nonnull String pem, @Nullable String passphrase) throws IOException {
+        decodePEM(pem, passphrase);
+    }
+
+    /**
+     *
+     * Creates a {@link PEMManager} from an {@link Object} that can be of any supported type by Bouncy Castle or JCA:
+     * {@link Key}, {@link PrivateKey}, {@link KeyPair}, {@link Certificate}, etc.
+     *
+     * @param pemObject object to manage
+     */
+    public PEMManager(@Nonnull Object pemObject) {
+        this.pemObject = pemObject;
+    }
+
+    private void decodePEM(@Nonnull File pemFile, @Nullable String passphrase) throws IOException {
+        decodePEM(FileUtils.readFileToString(pemFile), passphrase);
+    }
+
+    private void decodePEM(@Nonnull String pem, @Nullable final String passphrase) throws IOException {
+        PasswordFinder pwf = null;
+        if (passphrase != null) {
+            pwf = new PasswordFinder() {
+                @Override
+                public char[] getPassword() {
+                    return passphrase.toCharArray();
+                }
+            };
+        }
+
+        PEMReader parser = new PEMReader(new StringReader(pem), pwf);
+        try {
+            pemObject = parser.readObject();
+        } catch (PasswordException pwE) {
+            throw new BCPasswordException(pwE);
+        } finally {
+            parser.close();
+        }
+    }
+
+    /**
+     * Encodes the current stored information in PEM format and returns it as a {@link String}
+     * 
+     * @return PEM encoded data
+     * @throws IOException launched if a problem exists generating the PEM information
+     */
+    @Nonnull
+    public String encodePEM() throws IOException {
+        StringWriter sw = new StringWriter();
+        PEMWriter w = new PEMWriter(sw);
+        try {
+            w.writeObject(pemObject);
+        } finally {
+            w.close();
+        }
+        return sw.toString();
+    }
+
+    /**
+     * Encodes the current stored information in PEM formated {@link File}
+     * 
+     * @param pemFile PEM {@link File} to read
+     * 
+     * @throws IOException launched if a problem exists generating the PEM information or writing the {@link File}
+     */
+    public void encodePEM(@Nonnull File pemFile) throws IOException {
+        FileUtils.writeStringToFile(pemFile, encodePEM());
+    }
+
+    /**
+     * Obtain {@link KeyPair} object with the public and private key from the read PEM. No conversion is performed, the
+     * read PEM must contain private and public key in order to obtain a {@link KeyPair} object, null will be returned
+     * in all the other cases.
+     * 
+     * @return {@link KeyPair} object with public and private keys or null if the read PEM didn't contain private and
+     * public keys.
+     */
+    @Nullable
```

Comment:
this is `@CheckForNull` not `@Nullable` as the caller does not know before hand whether the PEM encoded object is a certificate or a key... thus *a priori* we do not know whether this could return `null` or `non-null` so the caller cannot be expected to know the null-ness and this means it cannot be `@Nullable` and must be `@CheckForNull`